simln-lib/src/test_utils.rs

```diff
@@ -45,3 +54,187 @@ pub fn create_nodes(n: usize, node_capacity: u64) -> Vec<(NodeInfo, u64)> {
         })
         .collect()
 }
+
+mock! {
+    pub Generator {}
+
+    impl fmt::Display for Generator {
+        fn fmt<'a>(&self, f: &mut fmt::Formatter<'a>) -> fmt::Result;
+    }
+
+    impl PaymentGenerator for Generator {
+        fn payment_start(&self) -> Option<Duration>;
+        fn payment_count(&self) -> Option<u64>;
+        fn next_payment_wait(&self) -> Result<Duration, PaymentGenerationError>;
+        fn payment_amount(&self, destination_capacity: Option<u64>) -> Result<u64, PaymentGenerationError>;
+    }
+}
+
+mock! {
+    pub LightningNode {}
+    #[async_trait]
+    impl crate::LightningNode for LightningNode {
+        fn get_info(&self) -> &NodeInfo;
+        async fn get_network(&mut self) -> Result<bitcoin::Network, LightningError>;
+        async fn send_payment(
+                &mut self,
+                dest: bitcoin::secp256k1::PublicKey,
+                amount_msat: u64,
+            ) -> Result<lightning::ln::PaymentHash, LightningError>;
+        async fn track_payment(
+                &mut self,
+                hash: &lightning::ln::PaymentHash,
+                shutdown: triggered::Listener,
+            ) -> Result<crate::PaymentResult, LightningError>;
+        async fn get_node_info(&mut self, node_id: &PublicKey) -> Result<NodeInfo, LightningError>;
+        async fn list_channels(&mut self) -> Result<Vec<u64>, LightningError>;
+    }
+}
+
+/// Type alias for the result of setup_test_nodes
+type TestNodesResult = (
+    Vec<NodeInfo>,
+    HashMap<PublicKey, Arc<Mutex<dyn LightningNode>>>,
+);
+
+/// A builder for creating mock Lightning nodes for testing purposes.
+///
+/// This struct provides a flexible way to configure test nodes with features like
+/// keysend support and specific networks. It supports both full node setups (with
+/// node info and clients) and client-only setups for network testing.
+///
+/// # Examples
+/// Basic setup with keysend on specific nodes:
+/// let (nodes, clients) = TestNetworkBuilder::setup_test_nodes(5, &[0, 2]);
```

Comment:
I think it is necessary to update the `TestNetworkBuilder` reference for `LightningTestNodeBuilder`